tarn/location/fanout.py

```diff
@@ -19,39 +19,36 @@ def __init__(self, *locations: Location):
 
     @contextmanager
     def read(self, key: Key, return_labels: bool) -> ContextManager[Union[None, Value, Tuple[Value, MaybeLabels]]]:
-        raised = False
         for location in self._locations:
+            leave = False
             with location.read(key, return_labels) as value:
                 if value is not None:
-                    try:
-                        yield value
-                        return
-                    except BaseException:
-                        raised = True
-                        raise
-
-            if raised:
+                    leave = True
+                    yield value
+
+            # see more info on the "leave" trick in `Levels`
+            if leave:
                 return
 
         yield None
 
     @contextmanager
     def write(self, key: Key, value: Value, labels: MaybeLabels) -> ContextManager[MaybeValue]:
-        raised = False
         for location in self._locations:
             if isinstance(location, Writable):
+                leave = False
                 with location.write(key, value, labels) as written:
                     if written is not None:
-                        try:
-                            yield written
-                            return
-                        except BaseException:
-                            raised = True
-                            raise
-
-                if raised:
+                        leave = True
+                        yield written
+
+                # see more info on the "leave" trick in `Levels`
+                if leave:
                     return
 
+                if isinstance(value, BinaryIO):
+                    value.seek(0)
```

Comment:
should take the `tell` value saved before writing, instead of going to 0. maybe write a context manager for that? e.g. 
```
with reusable(value):
   ...
```